src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_CAPACITY = 10;
+    private static final double MULTIPLICATION = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[ARRAY_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index"
+                    + index + " out of bounds exception.");
+        }
+        if (size == array.length) {
+            array = reSize(array);
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheck(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index);
+        T removedElement = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index + 1);
```

Comment:
should we do `size - index + 1` with the last parameter?

Reply:
Maybe no)